File: tests/humanize-test.py
#!/usr/bin/env python3
"""
Test humanize package - Python native version
"""

import humanize

# Test intcomma - format numbers with commas
print("=== Intcomma Tests ===")
print(f"intcomma(100) = {humanize.intcomma(100)}")
print(f"intcomma(1000) = {humanize.intcomma(1000)}")
print(f"intcomma(1000000) = {humanize.intcomma(1000000)}")
print(f"intcomma(1234567.25) = {humanize.intcomma(1234567.25)}")

# Test intword - convert large numbers to words
print("\n=== Intword Tests ===")
print(f"intword(100) = {humanize.intword(100)}")
print(f"intword(12400) = {humanize.intword(12400)}")
print(f"intword(1000000) = {humanize.intword(1000000)}")
print(f"intword(1200000000) = {humanize.intword(1200000000)}")

# Test naturalsize - format file sizes
print("\n=== Naturalsize Tests ===")
print(f"naturalsize(300) = {humanize.naturalsize(300)}")
print(f"naturalsize(3000) = {humanize.naturalsize(3000)}")
print(f"naturalsize(3000000) = {humanize.naturalsize(3000000)}")
print(f"naturalsize(3000000000) = {humanize.naturalsize(3000000000)}")

# Test ordinal - convert to ordinal
print("\n=== Ordinal Tests ===")
print(f"ordinal(1) = {humanize.ordinal(1)}")
print(f"ordinal(2) = {humanize.ordinal(2)}")
print(f"ordinal(3) = {humanize.ordinal(3)}")
print(f"ordinal(11) = {humanize.ordinal(11)}")
print(f"ordinal(111) = {humanize.ordinal(111)}")
print(f"ordinal(1002) = {humanize.ordinal(1002)}")

# Test apnumber - Associated Press style
print("\n=== Apnumber Tests ===")
print(f"apnumber(0) = {humanize.apnumber(0)}")
print(f"apnumber(5) = {humanize.apnumber(5)}")
print(f"apnumber(9) = {humanize.apnumber(9)}")
print(f"apnumber(10) = {humanize.apnumber(10)}")

# Test fractional - convert to fractions
print("\n=== Fractional Tests ===")
print(f"fractional(0.5) = {humanize.fractional(0.5)}")
print(f"fractional(0.3) = {humanize.fractional(0.3)}")
print(f"fractional(1.3) = {humanize.fractional(1.3)}")
print(f"fractional(1) = {humanize.fractional(1)}")

# Test scientific - scientific notation
# Note: explicitly passing precision=2 to match default
print("\n=== Scientific Tests ===")
print(f"scientific(500, 2) = {humanize.scientific(500, 2)}")
print(f"scientific(0.3, 2) = {humanize.scientific(0.3, 2)}")
print(f"scientific(-1000, 2) = {humanize.scientific(-1000, 2)}")

# Test metric - metric SI unit-prefix
# Note: explicitly passing precision=3 to match default
print("\n=== Metric Tests ===")
print(f"metric(1500, 'V', 3) = {humanize.metric(1500, 'V', 3)}")
print(f"metric(2e8, 'W', 3) = {humanize.metric(2e8, 'W', 3)}")
print(f"metric(220e-6, 'F', 3) = {humanize.metric(220e-6, 'F', 3)}")

# Test natural_list - natural list formatting
print("\n=== Natural List Tests ===")
print(f"natural_list(['one', 'two', 'three']) = {humanize.natural_list(['one', 'two', 'three'])}")
print(f"natural_list(['one', 'two']) = {humanize.natural_list(['one', 'two'])}")
print(f"natural_list(['one']) = {humanize.natural_list(['one'])}")

print("\n=== All tests completed ===")

File: tests/humanize-test.mjs
/**
 * Test humanize package - Node.js WASM version
 */

import {
  init,
  intcomma,
  intword,
  naturalsize,
  ordinal,
  apnumber,
  // fractional, // SKIPPED: 'fractions' module not available in WASM
  scientific,
  metric,
  naturalList
} from '../umo_modules/humanize/index.js';

await init();

// Test intcomma - format numbers with commas
console.log("=== Intcomma Tests ===");
console.log(`intcomma(100) = ${intcomma(100)}`);
console.log(`intcomma(1000) = ${intcomma(1000)}`);
console.log(`intcomma(1000000) = ${intcomma(1000000)}`);
console.log(`intcomma(1234567.25) = ${intcomma(1234567.25)}`);

// Test intword - convert large numbers to words
console.log("\n=== Intword Tests ===");
console.log(`intword(100) = ${intword(100)}`);
console.log(`intword(12400) = ${intword(12400)}`);
console.log(`intword(1000000) = ${intword(1000000)}`);
console.log(`intword(1200000000) = ${intword(1200000000)}`);

// Test naturalsize - format file sizes
console.log("\n=== Naturalsize Tests ===");
console.log(`naturalsize(300) = ${naturalsize(300)}`);
console.log(`naturalsize(3000) = ${naturalsize(3000)}`);
console.log(`naturalsize(3000000) = ${naturalsize(3000000)}`);
console.log(`naturalsize(3000000000) = ${naturalsize(3000000000)}`);

// Test ordinal - convert to ordinal
console.log("\n=== Ordinal Tests ===");
console.log(`ordinal(1) = ${ordinal(1)}`);
console.log(`ordinal(2) = ${ordinal(2)}`);
console.log(`ordinal(3) = ${ordinal(3)}`);
console.log(`ordinal(11) = ${ordinal(11)}`);
console.log(`ordinal(111) = ${ordinal(111)}`);
console.log(`ordinal(1002) = ${ordinal(1002)}`);

// Test apnumber - Associated Press style
console.log("\n=== Apnumber Tests ===");
console.log(`apnumber(0) = ${apnumber(0)}`);
console.log(`apnumber(5) = ${apnumber(5)}`);
console.log(`apnumber(9) = ${apnumber(9)}`);
console.log(`apnumber(10) = ${apnumber(10)}`);

// Test fractional - convert to fractions
// SKIPPED: 'fractions' module not available in WASM
console.log("\n=== Fractional Tests (SKIPPED - fractions module not available in WASM) ===");
// console.log(`fractional(0.5) = ${fractional(0.5)}`);
// console.log(`fractional(0.3) = ${fractional(0.3)}`);
// console.log(`fractional(1.3) = ${fractional(1.3)}`);
// console.log(`fractional(1) = ${fractional(1)}`);

// Test scientific - scientific notation
// Note: explicitly passing precision=2 to match Python default
console.log("\n=== Scientific Tests ===");
console.log(`scientific(500, 2) = ${scientific(500, 2)}`);
console.log(`scientific(0.3, 2) = ${scientific(0.3, 2)}`);
console.log(`scientific(-1000, 2) = ${scientific(-1000, 2)}`);

// Test metric - metric SI unit-prefix
// Note: explicitly passing precision=3 to match Python default
console.log("\n=== Metric Tests ===");
console.log(`metric(1500, 'V', 3) = ${metric(1500, 'V', 3)}`);
console.log(`metric(2e8, 'W', 3) = ${metric(2e8, 'W', 3)}`);
console.log(`metric(220e-6, 'F', 3) = ${metric(220e-6, 'F', 3)}`);

// Test naturalList - natural list formatting
console.log("\n=== Natural List Tests ===");
console.log(`naturalList(['one', 'two', 'three']) = ${naturalList(['one', 'two', 'three'])}`);
console.log(`naturalList(['one', 'two']) = ${naturalList(['one', 'two'])}`);
console.log(`naturalList(['one']) = ${naturalList(['one'])}`);

console.log("\n=== All tests completed ===");
